mlflow/projects/__init__.py

```diff
@@ -193,12 +189,12 @@ def _fetch_project(uri, force_tempdir, version=None, git_username=None, git_pass
                           path of local projects (i.e. perform a no-op for local projects).
     """
     parsed_uri, subdirectory = _parse_subdirectory(uri)
-    use_temp_dst_dir = force_tempdir or not _is_local_path(uri)
-    dst_dir = tempfile.mkdtemp() if use_temp_dst_dir else urllib.parse.urlparse(uri).path
+    use_temp_dst_dir = force_tempdir or _GIT_URI_REGEX.match(parsed_uri)
+    dst_dir = tempfile.mkdtemp() if use_temp_dst_dir else parsed_uri
     if use_temp_dst_dir:
         eprint("=== Fetching project from %s into %s ===" % (uri, dst_dir))
```

Comment:
This logic is bug-prone in general in that we compute the `dst_dir` upfront and then use it across all project types (Git & local project). However, it does allow us to log the `dst_dir` before we attempt to fetch the project. One fix would be to allow individual project types to decide whether or not to use a tempdir (for clarity) at the cost of duplicating this logging message